### PR TITLE
E2E fixtures

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

refs #2293 #2294

One of the main reason our E2E tests are flaky and slow is because we depend a lot of the db seeder and the state of the DB for our tests to be successful. One of the main advantages of running in parallel is that those test can be run in any order and each test can create and cleanup without interfering with other tests. This PR aims to help with that.

I'm basing this PR on previous work done by @denik1981 and will expand fixtures for booking and event types for now.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] yarn test-e2e

### CODE REVIEW

**Review by zomars — COMMENTED** [review]
Test are passing and ready for review

File: apps/web/playwright/booking-pages.test.ts [hunk 1]
```diff
@@ -58,32 +57,24 @@ test.describe("free user", () => {
...
-    await page.goto("/pro");
+  test.beforeEach(async ({ page, users }) => {
+    const pro = await users.create();
+    await page.goto(`/${pro.username}`);
   });
-
-  test.afterAll(async () => {
-    await deleteAllBookingsByEmail("pro@example.com");
+  test.afterEach(async ({ users }) => {
+    await users.deleteAll();
```
Comment (alannnc) [inline apps/web/playwright/booking-pages.test.ts:77]: Isn't this like the one above? It doesn't conflict with anything still being run?

File: apps/web/playwright/onboarding.test.ts [hunk 2]
```diff
@@ -30,6 +30,11 @@ test.describe("Onboarding", () => {
 
   test.describe("Onboarding", () => {
     test("update onboarding username via localstorage", async ({ page }) => {
+      /**
+       * We need to come up with a better test since all test are run in an incognito window.
+       * Meaning that all localstorage access is null here.
+       */
+      test.fixme();
```
[the remark this answers is not in the document]
Comment (alannnc) [reply apps/web/playwright/onboarding.test.ts:37]: noted

[the remark this answers is not in the document]
Comment (alannnc) [reply apps/web/playwright/onboarding.test.ts:37]: Taking a look..

**Review by alannnc — APPROVED** [review]
Code looks good, will test on my end soon.

File: apps/web/playwright/booking-pages.test.ts (hunk 1 above)
Comment (zomars) [reply apps/web/playwright/booking-pages.test.ts:77]: The users fixture only contains the users that we're created by that specific worker for that specific test. So it's only deleting all users that were created for this test.

Comment (alannnc) [reply apps/web/playwright/booking-pages.test.ts:77]: Got it!

**Pull request merged**